PDFlex/pdflex-backend
# Implemented Retrieve Form Use Case

## Motivation and Context


This pull request implements the Retrieve Form Use Case.

This retrieves the information regarding a form that is stored in a database so that the front-end can use it to pre-populate the HTML form. Note that this use case is trigged at two points:
a) Immediately following the upload use case, this use case populates the HTML form with the information retrieved from the OCR call for the user to confirm.
b) When the user selects Fill Out Form, in the case that the form, this use case is called, but will populate the form with empty values. As part of our stretch goals, this will eventually be expanded to allow users to revisit forms multiple times before submitting them officially. 


closes #43 

## Your Changes


Implemented all the CA architecture aspects for the use case. This also included a new method in our DataAccessObject, which retrieves the first form of a claim. 

**Type of change** (select all that apply):


- [X] Refactoring (internal change to codebase, without changing functionality)

## Notes for Reviewer


Note that this code has been tested through the front-end (end-to-end tested), and that Junit tests have yet to be written.
To test using front-end:
- Run the 'retrieve-form-use-case' branch on backend.
- Navigate to the 'retrieve-form-tester' branch on frontend and type 'npm start' in terminal to start the React Application.
- Navigate to http://localhost:3000
- Login as User 1234, proceed to create a new claim, and either of the Upload or Fill Out Form selections.
- You should currently see "Toby, Quinlan" as the name in the first input box to confirm that the use case is retrieving information correctly.

*Note that as of writing this PR request, the frontend is hard-coded to view claim 1001 regardless of the user login.

## Questions and Comments (if applicable)

## Checklist

- [X] have performed a self-review of my own code.
